### PR TITLE
feat(checks): New iam_securityaudit_role_created

### PR DESCRIPTION
**Pull request by gabrielsoltz**
### Context

This is a new IAM check: `iam_securityaudit_role_created`
Description: Ensure a Security Audit role has been created to conduct security audits

### Description

Creating an IAM user with a security audit role provides a clear separation of duties between the security team and other teams within the organization. This helps to ensure that security-related activities are performed by authorized individuals with the appropriate expertise and access permissions.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by MrCloudSec — COMMENTED** [review]
Awesome job! 
I would change the check name to iam_role_securityaudit_created so we follow the pattern service_subservice_etc...

**Comment by gabrielsoltz** [comment]
@sergargar i used this as reference: https://github.com/prowler-cloud/prowler/tree/master/prowler/providers/aws/services/iam/iam_support_role_created

File: prowler/providers/aws/services/iam/iam_service.py [hunk 1]
```diff
@@ -353,6 +356,20 @@ def __get_entities_attached_to_support_roles__(self):
...
+            securityaudit_roles = self.client.list_entities_for_policy(
+                PolicyArn=securityaudit_entry_policy_arn, EntityFilter="Role"
+            )["PolicyRoles"]
+        except Exception as error:
+            logger.error(
+                f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+        finally:
+            return securityaudit_roles
+
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_service.py]: Could you merge this function with the `__get_entities_attached_to_support_roles__` and pass the Role ARN as an argument?

Maybe you don't need neither of that and only use the `list_entities_for_policy`

File: tests/providers/aws/services/iam/iam_securityaudit_role_created/iam_securityaudit_role_created_test.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                new=IAM(audit_info),
+            ):
+                # Test Check
+                from prowler.providers.aws.services.iam.iam_securityaudit_role_created.iam_securityaudit_role_created import (
+                    iam_securityaudit_role_created,
+                )
+
+                check = iam_securityaudit_role_created()
+                result = check.execute()
+                assert result[0].status == "PASS"
```
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_securityaudit_role_created/iam_securityaudit_role_created_test.py:75]: Can you assert also the finding region?

File: tests/providers/aws/services/iam/iam_securityaudit_role_created/iam_securityaudit_role_created_test.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                )
+
+                check = iam_securityaudit_role_created()
+                result = check.execute()
+                assert result[0].status == "FAIL"
+                assert (
+                    result[0].status_extended
+                    == "SecurityAudit policy is not attached to any role"
+                )
+                assert result[0].resource_id == "SecurityAudit"
```
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_securityaudit_role_created/iam_securityaudit_role_created_test.py:108]: Can you assert also the finding region?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Good job @gabrielsoltz, could you please check my comments then it'll be ready to merge 🚀

File: prowler/providers/aws/services/iam/iam_service.py (hunk 1 above)
Comment (gabrielsoltz) [reply prowler/providers/aws/services/iam/iam_service.py]: I merged both functions into only one with an argument, I also took the freedom to rename the lists we are using for these checks, as the names were incorrect... These are entities (roles) for a policy, not for a role, the role is the entity.

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**